src/action/SummationAction.cpp

```diff
@@ -0,0 +1,44 @@
+#include "SummationAction.hpp"
+#include "mesh/Data.hpp"
+#include "mesh/Edge.hpp"
+#include "mesh/Mesh.hpp"
+
+namespace precice {
+namespace action{
+
+SummationAction::SummationAction(
+	Timing				timing,
+	std::vector<int>	sourceDataID,
+	int					targetDataID,
+	const mesh::PtrMesh	&mesh)
+	: Action(timing, mesh, mapping::Mapping::MeshRequirement::VERTEX), _targetData(mesh->data(targetDataID))
+{
+
+	for(auto sourceID : sourceDataID){
+		_sourceData.push_back(mesh->data(sourceID));
+	}
+
+	for(const auto & source : _sourceData) {
+		PRECICE_ASSERT(source->getDimensions() == _targetData->getDimensions(), source->getDimensions(), _targetData->getDimensions());
```

Comment:
We use `PRECICE_ASSERT` for assertions, so to safeguard implementation bugs. `PRECICE_CHECK`, on the other hand checks whether user input was correct. And if not we through an error. I guess here, we actually need a `PRECICE_CHECK` (it might also be wrong in other implemented actions). [Here](https://github.com/precice/precice/blob/develop/src/precice/impl/SolverInterfaceImpl.cpp#L558-L559) is an example.